src/lib.rs

```diff
@@ -130,6 +130,182 @@ impl<'a> Drop for Sentinel<'a> {
+    ///
+    /// [`ThreadPool`]: struct.ThreadPool.html
+    ///
+    /// # Panics
+    ///
+    /// This method will panic if `num_threads` is 0.
+    ///
+    /// # Examples
+    ///
+    /// No more than eight threads will be alive simultaneously for this pool:
+    ///
+    /// ```
+    /// use std::thread;
+    ///
+    /// let pool = threadpool::Builder::new()
+    ///     .num_threads(8)
+    ///     .finish();
+    ///
+    /// for _ in 0..100 {
+    ///     pool.execute(|| {
+    ///         println!("Hello from a worker thread!")
+    ///     })
+    /// }
+    /// ```
+    pub fn num_threads(mut self, num_threads: usize) -> Builder {
+        assert!(num_threads > 0);
+        self.num_threads = Some(num_threads);
+        self
+    }
+
+    /// Set the thread name for each of the threads spawned by the built [`ThreadPool`]. If not
+    /// specified, threads spawned by the thread pool will be unnamed.
+    ///
+    /// [`ThreadPool`]: struct.ThreadPool.html
+    ///
+    /// # Examples
+    ///
+    /// Each thread spawned by this pool will have the name "foo":
+    ///
+    /// ```
+    /// use std::thread;
+    ///
+    /// let pool = threadpool::Builder::new()
+    ///     .thread_name("foo".into())
+    ///     .finish();
+    ///
+    /// for _ in 0..100 {
+    ///     pool.execute(|| {
+    ///         assert_eq!(thread::current().name(), Some("foo"));
+    ///     })
+    /// }
+    /// ```
+    pub fn thread_name(mut self, name: String) -> Builder {
+        self.thread_name = Some(name);
+        self
+    }
+
+    /// Set the stack size (in bytes) for each of the threads spawned by the built [`ThreadPool`].
+    /// If not specified, threads spawned by the threadpool will have a stack size [as specified in
+    /// the `std::thread` documentation][thread].
+    ///
+    /// [thread]: https://doc.rust-lang.org/nightly/std/thread/index.html#stack-size
+    /// [`ThreadPool`]: struct.ThreadPool.html
+    ///
+    /// # Examples
+    ///
+    /// Each thread spawned by this pool will have a 4 MB stack:
+    ///
+    /// ```
+    /// let pool = threadpool::Builder::new()
+    ///     .thread_stack_size(4_000_000)
+    ///     .finish();
+    ///
+    /// for _ in 0..100 {
+    ///     pool.execute(|| {
+    ///         println!("This thread has a 4 MB stack size!");
+    ///     })
+    /// }
+    /// ```
+    pub fn thread_stack_size(mut self, size: usize) -> Builder {
+        self.thread_stack_size = Some(size);
+        self
+    }
+
+    /// Finalize the [`Builder`] and build the [`ThreadPool`].
+    ///
+    /// [`Builder`]: struct.Builder.html
+    /// [`ThreadPool`]: struct.ThreadPool.html
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let pool = threadpool::Builder::new()
+    ///     .num_threads(8)
+    ///     .thread_stack_size(4_000_000)
+    ///     .finish();
+    /// ```
+    pub fn finish(self) -> ThreadPool {
```

Comment:
Did you consider calling it `build` or `create`?

Reply:
I like `build`. I'll change it to that